docs/content/guides/dagster-pipes.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: Dagster Pipes | Dagster Docs
+description: "Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol."
+---
+
+# Dagster Pipes
+
+Dagster Pipes is a toolkit for building integrations between Dagster and external execution environments. It standardizes the process of passing parameters, injecting context information, ingesting logs, and collecting metadata all while remaining agnostic to how remote computations are launched in those environments. This enables the separation of orchestration and business logic in the Dagster ecosystem.
+
+It also smooths the process of incorporating pre-existing code, business logic, and execution environments into Dagster. Using Dagster Pipes, you add just a few lines to your code and it becomes seamlessly invocable by the orchestrator. In turn, you can stream logs and metadata back to Dagster so you can leverage its observability, lineage, cataloging, and debugging capabilities.
+
+---
+
+## Benefits
+
+With Dagster Pipes, you can:
+
+- Incorporate existing code into Dagster without huge refactors
+- Onboard stakeholder teams onto Dagster incrementally
+- Run code in external environments, and:
+  - Easily pass parameters to the code
+  - Stream unstructured logs and structured metadata back to Dagster
+- Separate orchestration and business logic environments
+- Use languages other than Python with Dagster
+
+---
+
+## Limitations
+
+While Dagster Pipes is lightweight and flexible, there are a few limitations to be aware of:
+
+- **Step launchers are currently incompatible.**
+- **I/O managers aren’t currently supported.** If you rely heavily on [I/O managers](/concepts/io-management/io-managers) in remote execution, we recommend using step launchers.
+
+---
+
+## How it works
+
+Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol.
+
+When Dagster Pipes is invoked, several steps will be carried out both in the **orchestration process**, or Dagster, and in the **external process**, such as Databricks.
+
+### In the orchestration process (Dagster)
+
+When the execution of a Dagster asset begins, Dagster makes context information (ex: `partition_key`, `asset_key`, etc.) available for the external process to consume.
+
+Next, Dagster launches the external process and passes in any provided parameters.
+
+<Image
+alt="Diagram explaining the Dagster Pipes process"
+src="/images/guides/dagster-pipes/dagster-pipes-process.png"
+width={1000}
+height={393}
+/>
+
+### In the external process
+
+The process starts and loads the context info provided by Dagster. When finished, the process sends execution data, logs, and any specified metadata, back to Dagster.
```

Comment:
> When finished, the process sends execution data, logs, and any specified metadata, back to Dagster.

This isn't 100% accurate. The information gets streamed back to Dagster through that "side channel" while the process is running, meaning the information (logs, metadata) can be sent back to Dagster even before the external process finishes.

Reply:
Any suggestions for how to word this? Given that we wanted to leave 'side channel' out of this page, I don't want to include it here.

Maybe?: "When the process is finished, execution data, logs, and any specified metadata will be sent back to Dagster." ?

Reply:
It typically happens continuously while the process is executing, not just sent on completion.

Reply:
> The process starts and loads the context info provided by Dagster. While the process runs, execution data, logs, and any specified metadata is streamed back to Dagster.